src/modules/peek/Peek.UI/MainWindowViewModel.cs

```diff
@@ -63,50 +126,124 @@ public void Initialize(HWND foregroundWindowHandle)
+
+        public void AttemptNextNavigation() => Navigate(NavigationDirection.Forwards);
+
+        private void Navigate(NavigationDirection direction, bool isAfterDelete = false)
         {
             if (NavigationThrottleTimer.IsEnabled)
             {
                 return;
             }
 
-            NavigationThrottleTimer.Start();
+            if (Items == null || Items.Count == _deletedItemIndexes.Count)
+            {
+                _currentIndex = DisplayIndex = 0;
+                CurrentItem = null;
+                return;
+            }
+
+            _navigationDirection = direction;
+
+            int offset = direction == NavigationDirection.Forwards ? 1 : -1;
+
+            do
+            {
+                _currentIndex = MathHelper.Modulo(_currentIndex + offset, Items.Count);
+            }
+            while (_deletedItemIndexes.Contains(_currentIndex));
+
+            CurrentItem = Items[_currentIndex];
+
+            // If we're navigating forwards after a delete operation, the displayed index does not
+            // change, e.g. "(2/3)" becomes "(2/2)".
+            if (isAfterDelete && direction == NavigationDirection.Forwards)
+            {
+                offset = 0;
+            }
+
+            DisplayIndex = MathHelper.Modulo(DisplayIndex + offset, DisplayItemCount);
 
-            var itemCount = Items?.Count ?? 1;
-            CurrentIndex = MathHelper.Modulo(CurrentIndex - 1, itemCount);
-            CurrentItem = Items?.ElementAtOrDefault(CurrentIndex);
+            NavigationThrottleTimer.Start();
         }
 
-        public void AttemptNextNavigation()
+        /// <summary>
+        /// Sends the current item to the Recycle Bin.
+        /// </summary>
+        public void DeleteItem()
         {
-            if (NavigationThrottleTimer.IsEnabled)
+            if (CurrentItem == null || !IsFilePath(CurrentItem.Path))
             {
                 return;
             }
 
-            NavigationThrottleTimer.Start();
+            _deletedItemIndexes.Add(_currentIndex);
+            OnPropertyChanged(nameof(DisplayItemCount));
+
+            string path = CurrentItem.Path;
+
+            DispatcherQueue.GetForCurrentThread().TryEnqueue(() =>
+            {
+                Task.Delay(DeleteDelayMs);
+                DeleteFile(path);
+            });
 
-            var itemCount = Items?.Count ?? 1;
-            CurrentIndex = MathHelper.Modulo(CurrentIndex + 1, itemCount);
-            CurrentItem = Items?.ElementAtOrDefault(CurrentIndex);
+            Navigate(_navigationDirection, isAfterDelete: true);
+        }
+
+        private void DeleteFile(string path, bool permanent = false)
+        {
+            SHFILEOPSTRUCT fileOp = new()
+            {
+                wFunc = FO_DELETE,
+                pFrom = path + "\0\0",
+                fFlags = (ushort)(FOF_NOCONFIRMATION | (permanent ? 0 : FOF_ALLOWUNDO)),
```

Comment:
Does this override the recycle bin property? It is possible to configure the delete confirmation in the recycle bin properties.

How does Fotos app handle it?

Reply:
This replicates how the Photos application handles deletes. Pressing Delete immediately sends the file to the Recycle Bin without popping up a message, and then it moves on to the next file.

The `permanent` argument is there for possible future use, but is always false for now.